### ISSUE TITLE
destroyVirtualMachine API call with expunge set to true returns null in the response

### ISSUE THREAD

**Issue by OlegChuev** [apache/cloudstack#9946]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.18.1.0
~~~

##### SUMMARY

When VM is destroyed with expunge = true, unexpected response is returned. 

##### STEPS TO REPRODUCE

~~~

1. Create any VM.
2. Click Destroy VM and set expunge equal to true.
3. Wait until the destroy process finishes.
4. Check the response.

~~~

![Screenshot 2024-11-19 at 12 28 14 PM](https://github.com/user-attachments/assets/fab117dc-1df5-459b-b311-69e56aeaa5ed)

![Screenshot 2024-11-19 at 12 36 23 PM](https://github.com/user-attachments/assets/f4645fc1-cf7e-405a-a2d1-9570343eff3d)

![Screenshot 2024-11-19 at 12 35 10 PM](https://github.com/user-attachments/assets/172a1a8b-20df-42e6-bfc6-94830802b385)

##### EXPECTED RESULTS

~~~
`destroyVirtualMachine` returns information about destroyed VM as it does for regular destroy
~~~

##### ACTUAL RESULTS

~~~
`destroyVirtualMachine` request with following params:

id: 353a646c-e91e-41c9-b916-ac3982a341bb
expunge: true
command: destroyVirtualMachine
response: json

returns unexpected jobresult:

{
    "queryasyncjobresultresponse": {
        "accountid": "41dcfb95-eda5-11ee-8e71-0242ac110002",
        "userid": "41dd1dfc-eda5-11ee-8e71-0242ac110002",
        "cmd": "org.apache.cloudstack.api.command.admin.vm.DestroyVMCmdByAdmin",
        "jobstatus": 1,
        "jobprocstatus": 0,
        "jobresultcode": 0,
        "jobresulttype": "object",
        "jobresult": {
            "null": {                                  // <- Why null? 
                "securitygroup": [],          // <- Why exactly these empty arrays?
                "nic": [],
                "affinitygroup": [],
                "tags": []
            }
        },
        "jobinstancetype": "VirtualMachine",
        "jobinstanceid": "353a646c-e91e-41c9-b916-ac3982a341bb",
        "created": "2024-11-19T10:29:02+0000",
        "completed": "2024-11-19T10:29:10+0000",
        "jobid": "9f623724-1d2c-4cf7-9cc2-5291e49e64a0"
    }
}
~~~

**Comment by DaanHoogland** [comment]
@OlegChuev , Can you test PR #9949? I think it addresses your issue.

**Comment by DaanHoogland** [comment]
fixed in #9949
